### DISCUSSION TITLE
how to populate form data using Controller

### DISCUSSION THREAD

**Discussion by bfate7** [react-hook-form/react-hook-form#6893] [Q&A] [answered]
im using material-ui and react-hook-form and im not able to populate form data from api using useEffect and setValue

react-hook-form v7.17.5
@hookform/resolvers v2.8.2
yup v0.32.11

codesandbox sample:
https://codesandbox.io/s/happy-sanderson-5wxz2?file=/src/App.js

**Comment by lundn** [comment]
I'm not sure why your case ain't working. In this scenario I would rather do the fetch first while showing a loader, then show the form with `defaultValues` set with the response when request is resolved.

**Comment by bfate7** [comment, marked as answer]
downgrade react-hook-form to v7.9.0 solves the issue

**Reply by bluebill1049** [reply]
you can also use reset as well for a whole form.
